src/test/fuzz/crypto.cpp

```diff
@@ -67,10 +77,11 @@ void test_one_input(const std::vector<uint8_t>& buffer)
             (void)sha1.Reset();
             (void)sha256.Reset();
             (void)sha512.Reset();
+            muhash = MuHash3072();
```

Comment:
Make sure you also exercise the other constructor `MuHash3072::MuHash3072(const unsigned char* key32)`. Perhaps `ConsumeBool()` to choose which constructor to use?

Generally a good thing to do when testing a fuzzer is to add `assert(false);` to all code paths you want to reach and then fuzz/tweak/repeat until you've reached them all.

In this case we want to make sure that relevant code paths in the following functions are covered:

```
MuHash3072::MuHash3072() noexcept
MuHash3072::MuHash3072(const unsigned char* key32) noexcept
void MuHash3072::Finalize(unsigned char* hash384) noexcept
MuHash3072& MuHash3072::operator*=(const MuHash3072& x) noexcept
MuHash3072& MuHash3072::operator/=(const MuHash3072& x) noexcept
```

Reply:
That constructor is used for the MuHash objects that get added or removed. I checked all the public functions as you suggested and they seem to all be covered.